lib/array_intersection.rb

```diff
@@ -1,6 +1,31 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), because the lookup for a hash is constant, and the times we do a lookup depends on the length of the longer array
+# Space complexity: O(n). where n is the length of the shortest array, since all elements of the shortest array are plaeced in the hash
+
 def intersection(array1, array2)
-  raise NotImplementedError
+  return [] unless array1 && array2
+  return [] if array1.length == 0 || array2.length == 0
+
+  if array1.length < array2.length
+    shorter_array = array1
+    longer_array = array2
+  else
+    shorter_array = array2
+    longer_array = array1
+  end
+
+  index = 0
+  shorter_array_hash = {}
+
+  until index > shorter_array.length
+    shorter_array_hash[shorter_array[index]] = 1
```

Comment:
I am not sure if you need a hash here. If the intersection of the arrays had multiple values that were in common (example: 
```
test1 = [1,2,3,4,5,5,5]
test2 = [5,5,5,5,5]
```
shorter_array_hash would be reset to a value of 1 each time for every iteration of 5. I think an array would work for your purposes.